python/src/iceberg/transforms.py

```diff
@@ -0,0 +1,583 @@
+
+    def preserves_order(self) -> bool:
+        return True
+
+    def satisfies_order_of(self, other: Transform) -> bool:
+        if self == other:
+            return True
+
+        if isinstance(other, DateTimeTransform):
+            return self._granularity.order <= other._granularity.order
+
+        return False
+
+    def to_human_string(self, value) -> str:
+        if value is None:
+            return "null"
+        return self._granularity.human_string(value)
+
+    def dedup_name(self) -> str:
+        return "time"
+
+
+class IdentityTransform(Transform):
+    def __init__(self, source_type: Type):
+        super().__init__(
+            "identity",
+            f"transforms.identity(source_type={repr(source_type)})",
+        )
+        self._type = source_type
+
+    def apply(self, value):
+        return value
+
+    def can_transform(self, target: Type) -> bool:
+        return target.is_primitive
+
+    def preserves_order(self) -> bool:
+        return True
+
+    def satisfies_order_of(self, other: Transform) -> bool:
+        return other.preserves_order()
+
+    def to_human_string(self, value) -> str:
+        if value is None:
+            return "null"
+        elif isinstance(self._type, DateType):
+            return transform_util.human_day(value)
+        elif isinstance(self._type, TimeType):
+            return transform_util.human_time(value)
+        elif isinstance(self._type, TimestampType):
+            return transform_util.human_timestamp(value)
+        elif isinstance(self._type, TimestamptzType):
+            return transform_util.human_timestamptz(value)
+        elif isinstance(self._type, FixedType):
+            return transform_util.base64encode(value)
+        elif isinstance(self._type, BinaryType):
+            return transform_util.base64encode(value)
+        else:
+            return str(value)
+
+
+class TruncateTransform(Transform):
+    """A transform for truncating a value to a specified width.
+
+    Args:
+      source_type (Type): An Iceberg Type of IntegerType, LongType, StringType, or BinaryType
+      width (int): The truncate width
+
+    Raises:
+      ValueError: If a type is provided that is incompatible with a Truncate transform
+    """
+
+    _VALID_TYPES = {IntegerType, LongType, StringType, BinaryType, DecimalType}
+
+    def __init__(self, source_type: Type, width: int):
+        if type(source_type) not in TruncateTransform._VALID_TYPES:
+            raise ValueError(f"Cannot truncate type: {source_type}")
+
+        super().__init__(
+            f"truncate[{width}]",
+            f"transforms.truncate(source_type={repr(source_type)}, width={width})",
+        )
+        self._type = source_type
+        self._width = width
+
+        if type(self._type) in {IntegerType, LongType}:
+            self._apply = lambda v, w: v - v % w
+        elif type(self._type) in {StringType, BinaryType}:
+            self._apply = lambda v, w: v[0 : min(w, len(v))]
```

Comment:
I think these two lambda functions would be better as module level functions in `transform_util.py`, mainly so that we can give them proper names and include docstrings that describe the arguments and expectations.

Reply:
SG, moved both into transform_util. Also, the other option is to add truncate method to types, then we don't need `if ... elif ... else` here.

Reply:
I tried both options and I think 2nd approach is better (by adding truncate into some of types classes). Updated the PR with this change and please take a look. Thanks.